teps/0091-trusted-resources.md

```diff
@@ -176,28 +162,29 @@ Task would still be considered verified.
     verification and image verification - for example, do I allow verified
     Task Bundles if the Task within the Bundle does not have verified images?
 
-### User Experience (optional)
+### User Experience
 
 **Author publishing a Resource.**
 A Resource Author may want their Resource and its
 outputs to be trusted so that provenance attestations can be confidently made. The author will use their own private key to sign the Resource and signatures are stored in `Annotations` map of the Resource.
-The private keys are generated, used and stored by the author. Public keys then are stored as secrets or via URI stored in the configmap. Similar mechanism is used in [Chains](https://github.com/tektoncd/chains).
+The private keys are generated, used and stored by the author. Public keys then are stored at a CRD (Custom Resource Defination).
 
-**Verify the Resource via validating webhook.**
-The verification will be done in pipeline's admission webhook, by default the webhook will skip a Resource's verification. This can be configured from configmap.
+Situation A: Users want to use their own resources and make sure they are not tampered before using in Tekton. The user will need to choose a certain key tool (In the scope of this TEP we will leverage [sigstore](https://github.com/sigstore) to support different keys like cosign or KMS keys) and generate private&public key pair. The user will use the signing cli we provided to sign the yaml files and signature is stored in the `Annotations`.
 
-**Configure the Chains to skip the run if failed verificaiton.**
-Chains can be configured to not create a provenance attestation if the verification fails.
 
+Situation B: There is a central place (e.g. Tekton Catalog) to host community's resources and can be shared by other users. So one user can submit his signed resources and public keys, or rely on the central place's keys to sign the resources.
 
-### Performance (optional)
+Situation C: User A signs his tasks and store somewhere (e.g. Catalog) and user B wants to use the tasks from user A. How does user A pass the public keys to user B is out of the scope of this TEP. But we assume that the public key and resources should not be changed at the same time.
 
-For remote Resources such as OCI bundle, it will take time to fetch the resources. Resources that are not
-configured to be verified will have no change in performance.
+**Verify the Resource via validating webhook.**
+The verification will be done in pipeline's admission webhook, by default the webhook will skip a Resource's verification. This can be configured from configmap by users to decide whether skip the verification or not.
+
+During verification, the signature will be extracted from the resource, and will first check if public key exsits in the resource's taskrun/pipelinerun. If not then go to config map to use the default keys.
```

Comment:
| and will first check if public key exsits in the resource's taskrun/pipelinerun

This is no longer true, right?

Reply:
Yes, fixed!